### ISSUE TITLE
Ambiguous behavior while Inserting block (Using API)

### ISSUE THREAD

**Issue by InEase** [siyuan-note/siyuan#7996]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Can the issue be reproduced with the default theme (daylight/midnight)?

- [X] I was able to reproduce the issue with the default theme

### Could the issue be due to extensions?

- [X] I've ruled out the possibility that the extension is causing the problem.

### Describe the problem

Dear D, V and Siyuan Community:

I am developing a plugin which needs to insert memos（备注), and I'm using [/api/block/insertBlock](https://github.com/siyuan-note/siyuan/blob/master/API_zh_CN.md#%E6%8F%92%E5%85%A5%E5%9D%97), but it seems impossible to accomplish by inserting markdown, when i insert:

```markdown
正文^（备注文本）^
```

It becomes

![image](https://user-images.githubusercontent.com/88016816/231985578-dcad78d6-ccf5-4717-8300-ce1b35db8b2e.png)

It seems i cant insert memos by markdown.

---

Then, i used the dom mode of the same API to insert the block, but the Ambiguous behavior occurs:

This is the request body:

```json
{
  "dataType": "dom",
  "data": "<div data-node-id=\"20230414155335-4yxf6xe\" data-node-index=\"1\" data-type=\"NodeParagraph\" class=\"p\" updated=\"20230414160136\"><div contenteditable=\"true\" spellcheck=\"false\">aa<span data-type=\"text\" style=\"color: var(--b3-font-color4);\">aa</span>a</div><div class=\"protyle-attr\" contenteditable=\"false\">​</div></div>",
  "previousID": "20230414160610-jhn9vjs"
}
```

and when i **fire the request for many times**, it inserted some blocks **with the same attributes, including the data-node-id**

![image](https://user-images.githubusercontent.com/88016816/231986804-91a1f2f1-cf62-4e06-8496-95ac07088f92.png)

but it seems the structure of the file is complete, when i close-and-reopen the file, all the blocks are normal.

![Uploading image.png…]()

So I wonder whether this behavior is expected, and will the same block_id result in some Ambiguous behaviors?

Thank you !

### Expected result

The data_node_id and other attributes should be normal

### Version environment

```markdown
- Version: v2.8.4
- Operating System: Windows 10 
- Browser (if used):
```

### Log file

aa

**Comment by 88250** [comment]
Please try specifying a different ID and see if it works.

**Comment by InEase** [comment]
Yes, by specifying a different ID, siyuan kept the id as-is even after reopening the file

This is the dom tree after **Insert**
![image](https://user-images.githubusercontent.com/88016816/231989644-55b86424-0df6-4d32-a992-9931f5981807.png)

This is the dom tree after **reopen the file**
![Uploading image.png…]()

Seems that the front-end will take the dom inserted without further processing.

So I think the solution maybe is that I just need to specify a different ID for the block?

**Comment by 88250** [comment]
Yes, please use a different ID. For the specific ID generation rules, please refer to the relevant code.

**Comment by InEase** [comment]
Thank you! Your quick reply really surprised me !!! I think the problem is solved and maybe we can close the issue?

**Comment by 88250** [comment]
Thanks.
